zerver/views/auth.py

```diff
@@ -801,16 +805,21 @@ def redirect_to_misconfigured_ldap_notice(request: HttpRequest, error_type: int)
 def show_deactivation_notice(request: HttpRequest, next: str = "/") -> HttpResponse:
     realm = get_realm_from_request(request)
     if realm and realm.deactivated:
-        if realm.deactivated_redirect is not None:
-            # URL hash is automatically preserved by the browser.
-            # See https://stackoverflow.com/a/5283739
-            redirect_to = get_safe_redirect_to(next, realm.deactivated_redirect)
-            return HttpResponseRedirect(redirect_to)
-
         realm_data_scrubbed = RealmAuditLog.objects.filter(
             realm=realm, event_type=AuditLogEventType.REALM_SCRUBBED
         ).exists()
-        context = {"realm_data_deleted": realm_data_scrubbed}
+        context = {
+            "realm_data_deleted": realm_data_scrubbed,
+            "deactivated_redirect": realm.deactivated_redirect,
+        }
+
+        # If the redirect is in the same domain, do an automatic redirect.
+        if (
+            realm.deactivated_redirect is not None
+            and get_subdomain_from_hostname(realm.deactivated_redirect, None) is None
+        ):
+            redirect_to = get_safe_redirect_to(next, realm.deactivated_redirect)
+            context["auto_redirect_to"] = redirect_to
```

Comment:
I don't understand this logic. 

Testing with things like `do_change_realm_subdomain(get_realm("analytics"), "learn", acting_user=None)`, it doesn't work. I think the problem is in part that `get_subdomain_from_hostname` is assuming it's getting a hostname, not a full URL; so probably `urllib` can help with that.

But also, I think the check might be inverted? `get_subdomain_from_hostname` returns `None` if passed a totally unrelated URL, not the other way around.

Reply:
Oops, changed. Tested that this works now.